employee-rostering-frontend/src/ui/components/calendar/Schedule.tsx

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as React from 'react';
+import { Calendar, momentLocalizer, EventProps } from 'react-big-calendar'
+import moment from 'moment';
+import EventWrapper from './EventWrapper';
+
+import 'react-big-calendar/lib/css/react-big-calendar.css';
+import './ReactBigCalendarOverrides.css';
+
+export interface Props<T extends object> {
+  startDate: Date;
+  endDate: Date;
+  events: T[];
+  showAllDayCell?: boolean;
+  dateFormat?: (date: Date) => string;
+  startAccessor: (event: T) => Date;
+  endAccessor: (event: T) => Date;
+  titleAccessor: (event: T) => string;
+  addEvent: (start: Date, end: Date) => void;
+  eventStyle: (event: T) => { style?: React.CSSProperties; className?: string };
+  wrapperStyle: (event: T) => { style?: React.CSSProperties; className?: string; props?: {} };
```

Comment:
It seems the `props` are no longer used (please confirm & remove). If it's the case then all three `*Style()` functions have the same return type. Can you come up with a good name for it? It think even something very basic such as `StyleWrapper`, `StyleContainer`, whatever.. will make the code easier to understand and talk about (compare
- "*<...>* is a function returning `StyleContainer`." versus
- "*<...>* is a function returning an object... you know, the object with `style` and `className`").

Reply:
I just realized the return types are dictated by the `Calendar` component. So maybe it's a good idea to reuse its types, so:
```ts
eventStyle: (event: T) => EventPropGetter<T>;
dayStyle: (date: Date) => DayPropGetter;
```
not sure about `wrapperStyle` as its type is not "dictated" but I guess we could reuse one of the above.

Reply:
I don't think we will use any additional props Calendar may introduce to `EventPropGetter` and `DayPropGetter`. Additionally, I think we still want to use `style` and `className` if they changed the prop name in the future (with a change to convert it). I thinking of a `StyleSupplier<T>` type: `(params: T) => { style?: React.CSStyle;  className?: string; }`. (Note `StyleSupplier<T> = EventPropGetter<T>` and `StyleSupplier<Date> = DayPropGetter`; I also think the name `StyleSupplier<T>` is more clear).